packages/pigeon/lib/objc_generator.dart

```diff
@@ -376,9 +380,13 @@ String _makeObjcSignature({
       _getSelectorComponents(func, lastArgName);
   final Iterable<String> argTypes = followedByOne(
     func.arguments.map((NamedType arg) {
-      final String nullable = arg.type.isNullable ? 'nullable ' : '';
-      final _ObjcPtr argType = _objcTypeForDartType(options.prefix, arg.type);
-      return '$nullable${argType.ptr.trim()}';
+      if (isEnum(arg.type)) {
+        return _className(options.prefix, arg.type.baseName);
```

Comment:
Doesn't an enum need to be wrapped in an NSNumber so that it can be nullable, like other primitives? (And shouldn't that logic be in `_objcTypeForDartType`?)

Reply:
We are blocking nullable enums for now so it isn't necessary.  When we support nullable enums that would be a good way to implement them.

Reply:
Sorry, I hadn't fully paged this back in when I re-reviewed. Even though you aren't supporting nullable enums now, we will want to add that support soon (since it's a weird gap), so it should be an NSNumber now otherwise it will be a breaking change to add nullable support.

Reply:
I was thinking I'd use `nullable NSNumber*` for nullable enums and the primitive type for non-null enums.  That way the usage is cleaner, safer and it matches the behavior we have for enums in class fields.  I don't think nullable enums are that high priority anyways since the null value can easily be represented with an enum field.  That would be preferable to losing the type safety of dealing directly with enums versus introducing nsnumbers.

Reply:
I didn't realize that's what we were doing for enum class fields, I thought they were boxed as well. The inconsistency with other primitive types, which we box regardless of whether or not they are nullable, seems odd, but it sounds like that ship has already sailed. Maybe at some point we can do a breaking change to better unify handling if we start doing switching on enums.